virtual_rainforest/models/abiotic/radiation.py

```diff
@@ -0,0 +1,215 @@
+            (SPLASH v.1.0): robust indices of radiation, evapotranspiration and plant-
+            available moisture, Geosci. Model Dev., 10, 689-708
+        """
+        # calculate transmissivity (tau), unitless
+        tau_o = CLOUDY_TRANSMISSIVITY + TRANSMISSIVITY_COEFFICIENT * sunshine_hours
+        tau = tau_o * (1.0 + BEER_REGRESSION * self.elevation)
+
+        # Calculate daily photosynth. photon flux density (ppfd_d), mol/m^2
+        self.ppfd = (1.0e-6) * FLUX_TO_ENERGY * (1.0 - albedo_vis) * tau * shortwave_in
+        return tau
+
+    def calc_topofcanopy_radiation(
+        self,
+        tau: NDArray[np.float32],
+        shortwave_in: NDArray[np.float32],
+        sunshine_hours: NDArray[np.float32] = np.array(1.0, dtype=np.float32),
+        albedo_shortwave: NDArray[np.float32] = ALBEDO_SHORTWAVE,
+    ) -> None:
+        """Calculate top of canopy shortwave radiation [J m-2].
+
+        Args:
+            tau: atmospheric transmissivity, unitless
+            shortwave_in: daily downward shortwave radiation [J m-2]
+            sunshine_hours: fraction of sunshine hours, between 0 (100% cloud cover)
+                and 1 (cloud free sky)
+            albedo_shortwave: shortwave albedo, default = 0.17
+        """
+        self.topofcanopy_radiation = (1.0 - albedo_shortwave) * tau * shortwave_in
+
+    def calc_longwave_radiation(
+        self,
+        canopy_temperature: NDArray[np.float32],
+        surface_temperature: NDArray[np.float32],
+    ) -> None:
+        """Calculates longwave emission from canopy and forest floor [J m-2].
+
+        Args:
+            canopy_temperature: canopy temperature of n layers [C]; the array size is
+                set to max number of layers (n_max) and filled with NaN where n < n_max
+            surface_temperature: surface soil temperature [C]
+        """
+        # longwave emission canopy
+        self.longwave_canopy = (
+            CANOPY_EMISSIVITY
+            * BOLZMAN_CONSTANT
+            * (CELSIUS_TO_KELVIN + canopy_temperature) ** 4
+        )
+
+        # longwave emission surface
+        self.longwave_soil = (
+            SOIL_EMISSIVITY
+            * BOLZMAN_CONSTANT
+            * (CELSIUS_TO_KELVIN + surface_temperature) ** 4
+        )
+
+    def calc_netradiation_surface(
+        self,
+        canopy_absorption: NDArray[np.float32],
+    ) -> None:
+        """Calculates daily net radiation at the forest floor [J m-2].
+
+        Args:
+            canopy_absorption: absorption by canopy layers, [J m-2]
+        """
+        self.netradiation_surface = (
+            self.topofcanopy_radiation
+            - canopy_absorption  # np.sum(canopy_absorption, axis=1)  # what axis?
+            - self.longwave_soil
+            - self.longwave_canopy  # np.sum(self.longwave_canopy, axis=1)
+        )
+
+    def radiation_balance(
+        self,
+        elevation: NDArray[np.float32],
```

Comment:
This is passed as an argument but is not used as `calc_ppfd` reads it directly from `self`.